[BUG] Mailbox contents not deleted

# Bug Report

## Context
Mailbox contents remain in `maildata` subdirectory after running `sudo ./setup.sh email del test@example.com`.

### What is affected by this bug?
Deleting mailboxes.
### When does this occur?
When deleting a mailbox with the `setup.sh` script.
### How do we replicate the issue?

```
sudo ./setup.sh email del test@example.com
Password: 
sed: can't read /tmp/docker-mailserver/postfix-virtual.cf: No such file or directory
Do you want to delete the mailbox as well(all mails will be removed)?(y/n) y

Mailbox deleted.
```
Check contents of `maildata` subdirectory.

## Behavior

### Actual Behavior
The mailbox is no longer listed when running `sudo ./setup.sh email list`. But the `test` mailbox and its contents are still in the `maildata` subdirectory. 
### Expected Behavior
The mailbox contents should have been deleted.
## Your Environment
Synology DS918+ with latest DSM and Docker version 18.09.8, build bfed4f5.

### Environment Variables

```yml
---
version: "2"

# Define services
services:
  
  roundcubemail:
    image: roundcube/roundcubemail:latest
    container_name: roundcubemail
    restart: unless-stopped
    volumes:
      - /v/d/roundcube/www:/var/www/html
      - /v/d/roundcube/sqlite:/var/roundcube/db
    ports:
      - 127.0.0.1:8000:80
    environment:
      - ROUNDCUBEMAIL_DB_TYPE=sqlite
      - ROUNDCUBEMAIL_SKIN=elastic
      - ROUNDCUBEMAIL_DEFAULT_HOST=ssl://mail.example.com
      - ROUNDCUBEMAIL_SMTP_SERVER=ssl://mail.example.com
      - ROUNDCUBEMAIL_DEFAULT_PORT=993
      - ROUNDCUBEMAIL_SMTP_PORT=465
  
  solr:
    container_name: mailserver_solr
    image: lmmdock/dovecot-solr:latest
    volumes:
      - /v/d/solr-dovecot:/opt/solr/server/solr/dovecot
    restart: always
  
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    hostname: mail
    domainname: example.com
    container_name: mailserver
    ports:
      - "465:465"
      - "993:993"
    volumes:
      - /v/d/mailserver/maildata:/var/mail
      - /v/d/mailserver/mailstate:/var/mail-state
      - /v/d/mailserver/maillogs:/var/log/mail
      - /v/d/mailserver/config:/tmp/docker-mailserver/
      - /v/d/ssl/output:/tmp/ssl:ro
      - /v/d/mailserver/solr-dovecot/10-plugin.conf:/etc/dovecot/conf.d/10-plugin.conf:ro
    environment:
      - SSL_TYPE=manual
      - SSL_CERT_PATH=/tmp/ssl/fullchain.pem
      - SSL_KEY_PATH=/tmp/ssl/privkey.pem
      - POSTFIX_MESSAGE_SIZE_LIMIT=102400000
      - ENABLE_SPAMASSASSIN=0
      - ENABLE_CLAMAV=0
      - ENABLE_FAIL2BAN=0
      - ENABLE_POSTGREY=0
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
      - DMS_DEBUG=0
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    restart: always
```
## Workaround?
Contents of mailbox are deleted when running `sudo docker exec mailserver rm -R /var/mail/example.com/test/` afterwards.

This:

``` BASH
#! /bin/bash

# ? This is done to ignore the message "Make sure not to read and write
# ? the same file in the same pipeline", which is a result of ${DATABASE}
# ? being used below. (This disables the message file-wide.)
# shellcheck disable=SC2094

# shellcheck source=../scripts/helper-functions.sh
. /usr/local/bin/helper-functions.sh

DATABASE=${DATABASE:-/tmp/docker-mailserver/postfix-accounts.cf}
ALIAS_DATABASE="/tmp/docker-mailserver/postfix-virtual.cf"
QUOTA_DATABASE="/tmp/docker-mailserver/dovecot-quotas.cf"

function usage
{
  echo "Usage: delmailuser [-y] <user@domain> <user2@anotherdomain> ..."
}

while getopts ":y" OPT
do
  case ${OPT} in
    y)
      MAILDEL="y"
      ;;
    \?)
      usage
      errex "Invalid option: -${OPTARG}"
      ;;
    *)
      usage
      errex "Invalid option: -${OPTARG}"
      ;;
  esac
done

shift $((OPTIND-1))

[[ -z ${*} ]] && { usage ; errex "No user specifed" ; }
[[ -s ${DATABASE} ]] || exit 0

(
  flock -e 200

  for USER in "${@}"
  do
    # very simple plausibility check
    [[ ${USER} != *'@'*'.'* ]] && errex "No valid address: ${USER}"

    declare -a MAILARR
    MAILARR[0]="${USER%@.*}"
    MAILARR[1]="${USER#.*@}"

    # ${USER} must not contain /s and other syntactic characters
    USER=$(escape "${USER}")

    if [[ -f ${DATABASE} ]]
    then
      if ! sed -i "/^""${USER}""|/d" "${DATABASE}"
      then
        errex "${USER} couldn't be deleted in ${DATABASE}. ${?}"
      fi
    fi

    if [[ -f ${ALIAS_DATABASE} ]]
    then
      # delete all aliases where the user is the only recipient( " ${USER$}" )
      # delete user only for all aliases that deliver to multiple recipients ( ",${USER}" "${USER,}" )
      if sed -i \
        -e "/ ""${USER}""$/d" -e "s/,""${USER}""//g" -e "s/""${USER}"",//g" \
        "${ALIAS_DATABASE}"
      then
        echo "${USER} and potential aliases deleted."
      else
        errex "Aliases for ${USER} couldn't be deleted in ${ALIAS_DATABASE}. ${?}"
      fi
    fi

    # remove quota directives
    if [[ -f ${QUOTA_DATABASE} ]]
    then
      if ! sed -i -e "/^${USER}:.*$/d" "${QUOTA_DATABASE}"
      then
        errex "Quota for ${USER} couldn't be deleted in ${QUOTA_DATABASE}. ${?}"
      fi
    fi

    if [[ ${MAILDEL} != "y" ]]
    then
      read -r -p "Do you want to delete the mailbox as well(all mails will be removed)?(y/n) " MAILDEL
    fi

    if [[ ${MAILDEL} != "y" ]]
    then
      echo "Leaving the mailbox untouched. If you want to delete it at a later point use 'sudo docker exec mail rm -R /var/mail/${MAILARR[1]}/${MAILARR[0]}'"
      exit 0
    fi

    if [[ -d "/var/mail/${MAILARR[1]}/${MAILARR[0]}" ]]
    then
      if rm -r -f "/var/mail/${MAILARR[1]}/${MAILARR[0]}"
      then
        echo "Mailbox deleted."
      else
        errex "Mailbox couldn't be deleted: ${?}"
      fi
    else
      errex "Mailbox directory '/var/mail/${MAILARR[1]}/${MAILARR[0]}' does not exist."
    fi
  done

) 200< "${DATABASE}"
```

is my refactoring. Seems a lot cleaner and more robust to me.

--- a/target/bin/delmailuser
+++ b/target/bin/delmailuser
@@ -1,9 +1,9 @@
 #! /bin/bash
 
+# shellcheck disable=SC2094
 # ? This is done to ignore the message "Make sure not to read and write
 # ? the same file in the same pipeline", which is a result of ${DATABASE}
 # ? being used below. (This disables the message file-wide.)
-# shellcheck disable=SC2094
 
 # shellcheck source=../scripts/helper-functions.sh
 . /usr/local/bin/helper-functions.sh
@@ -11,82 +11,98 @@
 DATABASE=${DATABASE:-/tmp/docker-mailserver/postfix-accounts.cf}
 ALIAS_DATABASE="/tmp/docker-mailserver/postfix-virtual.cf"
 QUOTA_DATABASE="/tmp/docker-mailserver/dovecot-quotas.cf"
+MAILDEL=false
 
 function usage
 {
-  echo "Usage: delmailuser <-y> <user@domain> <user2@anotherdomain> ..."
-  echo "	-y: don't prompt for confirmations"
+  echo "Usage: delmailuser [-y] <user@domain> [<user2@anotherdomain>]"
 }
 
-while getopts ":y" OPT
+while getopts ":yY" OPT
 do
   case ${OPT} in
-    y)
-      MAILDEL="y"
+    y | Y )
+      MAILDEL=true
       ;;
-    \?)
+    * )
       usage
-      errex "Invalid option: -${OPTARG}"
-      ;;
-    *)
-      usage
-      errex "Invalid option: -${OPTARG}"
+      exit 1
       ;;
   esac
 done
-
 shift $((OPTIND-1))
 
 [[ -z ${*} ]] && { usage ; errex "No user specifed" ; }
 [[ -s ${DATABASE} ]] || exit 0
 
-# Protect config file with lock to avoid race conditions
+if ! ${MAILDEL}
+then
+  read -r -p "Do you want to delete the mailbox as well (removing all mails) ? [Y/n] " MAILDEL_CHOSEN
+  [[ ${MAILDEL_CHOSEN} =~ (y|Y| ) ]] && MAILDEL=true
+fi
+
 (
   flock -e 200
 
   for USER in "${@}"
   do
     # very simple plausibility check
-    [[ ${USER} != *"@"*"."* ]] && errex "No valid address: ${USER}"
+    [[ ${USER} != *'@'*'.'* ]] && errex "No valid address: ${USER}"
 
     declare -a MAILARR
     MAILARR[0]="${USER%@*}"
     MAILARR[1]="${USER#*@}"
 
-    # XXX ${USER} must not contain /s and other syntactic characters
+    # ${USER} must not contain /s and other syntactic characters
     USER=$(escape "${USER}")
 
-    if ! sed -i "/^""${USER}""|/d" "${DATABASE}"
+    if [[ -f ${DATABASE} ]]
     then
-      errex "${USER} couldn't be deleted in ${DATABASE}. ${?}"
+      if ! sed -i "/^${USER}|/d" "${DATABASE}"
+      then
+        errex "${USER} couldn't be deleted in ${DATABASE}. ${?}"
+      fi
     fi
 
-    # Delete all aliases where the user is the only recipient( " ${USER$}" )
-    # Delete user only for all aliases that deliver to multiple recipients ( ",${USER}" "${USER,}" )
-    if sed -i -e "/ ""${USER}""$/d" \
-        -e "s/,""${USER}""//g" \
-        -e "s/""${USER}"",//g" "${ALIAS_DATABASE}"
+    if [[ -f ${ALIAS_DATABASE} ]]
     then
-      echo "${USER} and potential aliases deleted." || errex "Aliases for ${USER} couldn't be deleted in ${ALIAS_DATABASE}. ${?}"
+      # delete all aliases where the user is the only recipient( " ${USER$}" )
+      # delete user only for all aliases that deliver to multiple recipients ( ",${USER}" "${USER,}" )
+      if sed -i \
+        -e "/ ${USER}$/d" -e "s/,${USER}//g" -e "s/${USER},//g" \
+        "${ALIAS_DATABASE}"
+      then
+        echo "${USER} and potential aliases deleted."
+      else
+        errex "Aliases for ${USER} couldn't be deleted in ${ALIAS_DATABASE}."
+      fi
     fi
 
     # remove quota directives
     if [[ -f ${QUOTA_DATABASE} ]]
     then
-      sed -i -e "/^${USER}:.*$/d" "${QUOTA_DATABASE}" || errex "Quota for ${USER} couldn't be deleted in ${QUOTA_DATABASE}. ${?}"
+      if ! sed -i -e "/^${USER}:.*$/d" "${QUOTA_DATABASE}"
+      then
+        errex "Quota for ${USER} couldn't be deleted in ${QUOTA_DATABASE}."
+      fi
     fi
 
-    if [[ ${MAILDEL} != "y" ]]
+    if ! ${MAILDEL}
     then
-      read -r -p "Do you want to delete the mailbox as well(all mails will be removed)?(y/n) " MAILDEL
-      echo
+      echo "Leaving the mailbox untouched. If you want to delete it at a later point use 'sudo docker exec mail rm -R /var/mail/${MAILARR[1]}/${MAILARR[0]}'"
+      exit 0
     fi
 
-    [[ ${MAILDEL} != "y" ]] && errex "Leaving the mailbox untouched. If you want to delete it at a later point use \"sudo docker exec mail rm -R /var/mail/${MAILARR[1]}/${MAILARR[0]}\""
-
-    if rm -r -f "/var/mail/${MAILARR[1]}/${MAILARR[0]}"
+    if [[ -d "/var/mail/${MAILARR[1]}/${MAILARR[0]}" ]]
     then
-      echo "Mailbox deleted." || errex "Mailbox couldn't be deleted: ${?}"
+      if rm -r -f "/var/mail/${MAILARR[1]}/${MAILARR[0]}"
+      then
+        echo "Mailbox deleted."
+      else
+        errex "Mailbox couldn't be deleted."
+      fi
+    else
+      errex "Mailbox directory '/var/mail/${MAILARR[1]}/${MAILARR[0]}' does not exist."
     fi
   done
 

--- a/target/bin/delmailuser
+++ b/target/bin/delmailuser
@@ -102,7 +102,7 @@ fi
         errex "Mailbox couldn't be deleted."
       fi
     else
-      errex "Mailbox directory '/var/mail/${MAILARR[1]}/${MAILARR[0]}' does not exist."
+      echo "Mailbox directory '/var/mail/${MAILARR[1]}/${MAILARR[0]}' did not exist."
     fi
   done
 

--- a/target/scripts/start-mailserver.sh
+++ b/target/scripts/start-mailserver.sh
@@ -86,7 +86,6 @@ function register_functions
 
   ################### >> check funcs
 
-  _register_check_function "_check_environment_variables"
   _register_check_function "_check_hostname"
 
   ################### >> setup funcs
@@ -308,7 +307,7 @@ function check
 
 function _check_hostname
 {
-  _notify "task" "Check that hostname/domainname is provided or overridden (no default docker hostname/kubernetes) [in ${FUNCNAME[0]}]"
+  _notify "task" "Check that hostname/domainname is provided or overridden [in ${FUNCNAME[0]}]"
 
   if [[ -n ${OVERRIDE_HOSTNAME} ]]
   then
@@ -319,19 +318,11 @@ function _check_hostname
   _notify 'inf' "Domain has been set to ${DOMAINNAME}"
   _notify 'inf' "Hostname has been set to ${HOSTNAME}"
 
-  if ( ! grep -E '^(\S+[.]\S+)$' <<< "${HOSTNAME}" >/dev/null )
+  if ! grep -q -E '^(\S+[.]\S+)$' <<< "${HOSTNAME}"
   then
     _notify 'err' "Setting hostname/domainname is required"
     kill "$(< /var/run/supervisord.pid)" && return 1
-  else
-    return 0
   fi
-}
-
-function _check_environment_variables
-{
-  _notify "task" "Check that there are no conflicts with env variables [in ${FUNCNAME[0]}]"
-  return 0
 }
 
 ##########################################################################

